### PR TITLE
Add --network-dataplane to AKS cluster create

### PR DESCRIPTION
**Pull request by wedaly**
The `--network-dataplane` flag is used to enable Cilium dataplane.

This deprecates the existing `--enable-cilium-dataplane` flag, but for now keep both since the docs still refer to the old flag

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az aks create`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

~~For new extensions:~~

~~- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).~~

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

**Comment by wedaly** [comment]
keeping as draft until #6062 merges (those commits included in this branch)

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -271,6 +274,7 @@ def load_arguments(self, _):
         c.argument('network_plugin', arg_type=get_enum_type(network_plugins))
         c.argument('network_plugin_mode', arg_type=get_enum_type(network_plugin_modes))
         c.argument('network_policy')
+        c.argument('network_dataplane', arg_type=get_enum_type(network_dataplanes))
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:277]: If you don't mind, please set this new parameter many lines below as there gathers the options for aks-preview only.

Also please update the parameter `enable_cilium_dataplane` to set it as deprecated.

`c.argument('enable_cilium_dataplane', action='store_true', is_preview=True, deprecate_info=c.deprecate(target='--enable-cilium-dataplane', redirect='--network-dataplane', hide=True))`

Comment (wedaly) [reply src/aks-preview/azext_aks_preview/_params.py:277]: Updated to add `deprecate_info`

The `--network-dataplane` flag is going to be added to the stable CLI as well. Do you still want me to move it?

Comment (wedaly) [reply src/aks-preview/azext_aks_preview/_params.py:277]: Draft PR for the stable CLI: https://github.com/Azure/azure-cli/pull/25951

**Comment by wedaly** [comment]
rebased to resolve merge conflict

**Review by FumingZhang — APPROVED** [review]
LGTM

Queued a [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=71005720&view=results) to verify the newly added test case.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**